### PR TITLE
Enhance calibration of rolling shutter cameras (fixes #325, possibly #281)

### LINKED ISSUE #325: Camera calibration should ignore frames with large motion (mainly for rolling shutter)
Camera calibration takes images out of the video stream while the camera is moving. This means that slightly blurred images are used for calibration, which reduces the location accuracy of the corners points. Even worse, for rolling shutter cameras, the locations of corner points get shifted. This shift easily amounts to several pixels between the upper and the lower part of the pattern.

I noticed that calibration results differ a lot between several runs with rolling shutter cameras. A quick test revealed that this changes as soon as frames with large motion are rejected.

Did anyone see similar problems?

### PR DESCRIPTION
**Pull request by sevangelatos**
This pull request aims to improve the calibration accuracy of rolling shutter cameras by avoiding picking frames where the chessboard pattern is moving in the image frame. This should fix #325 .
This is controlled by the --max-chessboard-speed option that allows setting a maximum allowed speed in pixels/frame for the chessboard. If the chessboard is moving faster than the allowed speed, the frame is not picked to be used for calibration.

The chessboard speed is determined by measuring the average distance that each chessboard feature traveled between this and the previous frame.

### CODE REVIEW

**Comment by sevangelatos** [comment]
Any maintainers that can take a look please?

**Comment by JWhitleyWork** [comment]
@sevangelatos - I know this PR is a bit dated but would you mind rebasing your changes on `melodic` to pull in the new CI and other updates?

**Comment by sevangelatos** [comment]
@JWhitleyAStuff  I rebased it on the melodic branch. Lets see if this will fix the CI. I will re-test it this weekend and report back.

**Comment by JWhitleyWork** [comment]
@sevangelatos - I only just realized that the "Build Forked Pull Requests" option on CircleCI wasn't turned on. Can you push something innocuous to your branch (like a spacing change) to trigger CI?

**Review by JWhitleyWork — APPROVED** [review]

**Review by SteveMacenski — APPROVED** [review]
I have no comments worth blocking this for, but if you're experiencing high motion blur during calibration, I would *highly* recommend recalibrating the cameras. The quality of the calibration even by skipping frames above some thresholded value is questionable.

**Comment by JWhitleyWork** [comment]
@sevangelatos - Would you mind creating another PR back-porting this to the `kinetic` branch (making sure to update your `kinetic` branch with ours first)?

**Comment by sevangelatos** [comment]
@JWhitleyAStuff Sure, I will give it a try to backport this to kinetic. BTW, it seems to work fine on melodic after the merge.

**Comment by sevangelatos** [comment]
@JWhitleyAStuff I just created a back-port PR for kinetic. It's [here](https://github.com/ros-perception/image_pipeline/pull/415).

**Pull request merged**
